File: DevExpress-Demo/CS/TreeListDemo.Wpf/Data/DataAnnotations/DataAnnotationsElement2.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel;

namespace TreeListDemo {
    public class DataAnnotationsElement2 {

        [Display(AutoGenerateField = false)]
        public int ID { get; set; }

        [Display(AutoGenerateField = false)]
        public int ParentID { get; set; }

        [Editable(false)]
        public string Employer { get; set; }

        [Display(Name = "First name"), Required]
        public string FirstName { get; set; }

        [Display(Name = "Last name"), Required]
        public string LastName { get; set; }

        [Display(Name = "Full name")]
        public string FullName { get { return FirstName + " " + LastName; } }

        public Gender Gender { get; set; }

        public int Age { get; set; }

        [ReadOnlyAttribute(true)]
        public string SSN { get; set; }

        public override string ToString() {
            return "Attribute support";
        }
    }
}

File: DevExpress-Demo/VB/TreeListDemo.Wpf/Data/DataAnnotations/DataAnnotationsElement2.vb
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel

Namespace TreeListDemo

    Public Class DataAnnotationsElement2

        <Display(AutoGenerateField:=False)>
        Public Property ID As Integer

        <Display(AutoGenerateField:=False)>
        Public Property ParentID As Integer

        <Editable(False)>
        Public Property Employer As String

        <Display(Name:="First name"), Required>
        Public Property FirstName As String

        <Display(Name:="Last name"), Required>
        Public Property LastName As String

        <Display(Name:="Full name")>
        Public ReadOnly Property FullName As String
            Get
                Return FirstName & " " & LastName
            End Get
        End Property

        Public Property Gender As Gender

        Public Property Age As Integer

        <ReadOnlyAttribute(True)>
        Public Property SSN As String

        Public Overrides Function ToString() As String
            Return "Attribute support"
        End Function
    End Class
End Namespace
